### PR TITLE
CHE-1758 Display loading steps when booting ws from IDE

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Displays a new loader on top of the IDE

https://github.com/eclipse/che/issues/1758

Signed-off-by: Vitaliy Guliy vguliy@codenvy.com

### CODE REVIEW

**Comment by slemeur** [comment]
Please provide elements so that we can review the results: animated gif would be nice.

**Comment by vitaliy-guliy** [comment]
https://www.youtube.com/watch?v=fgecmMb7t9A

**Comment by TylerJewell** [comment]
@vitaliy-guliy - nice work!.  Can we make much more narrow the width in this graphic?  It needs to be wide enough to support machine names of about 15 characters, but not as wide as it is now.
capture

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package org.eclipse.che.ide.ui.loaders;
```
Comment (olexii4) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.java]: Add license

Comment (olexii4) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.java:11]: Add license

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderFactory.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package org.eclipse.che.ide.ui.loaders;
```
Comment (olexii4) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderFactory.java:11]: Add license

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * @author Vitaliy Guliy
+ */
+public class PopupLoaderImpl extends Composite implements PopupLoader {
+
+    interface LoaderPopupImplUiBinder extends UiBinder<FlowPanel, PopupLoaderImpl> {
+    }
+
+    // Loader title
```
Comment (vzhukovs) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.java:36]: useless comment

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.ui.xml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        .main {
+            width: 450px;
+            height: 57px;
+            position: fixed;
+            left: 50%;
+            margin-left: -225px;
+            top: 1px;
+            z-index: 99999;
+            background-color: #212325;
```
Comment (vzhukovs) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/PopupLoaderImpl.ui.xml]: declare colors in theme configuration

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineStatusNotifier.java [hunk 5]
```diff
@@ -98,8 +95,10 @@ public void apply(PromiseError arg) throws OperationException {
             @Override
             public void apply(MachineDto machine) throws OperationException {
                 if (machine.getConfig().isDev()) {
-                    initialLoadingInfo.setOperationStatus(MACHINE_BOOTING.getValue(), IN_PROGRESS);
+                    // Will be used later
```
Comment (vzhukovs) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineStatusNotifier.java:98]: remove commented code

**Comment by ashumilova** [comment]
make sure `mvn license:check` passes

File: ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/LoaderPresenter.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+ * Manages loaders for loading phases.
+ *
+ * @author Vitaliy Guliy
+ */
+@Singleton
+public class LoaderPresenter {
+
+    public enum Phase {
+
+        STARTING_WORKSPACE_RUNTIME("Starting workspace runtime", "Retrieving the stack's image and launching it"),
```
Comment (ashumilova) [inline ide/commons-gwt/src/main/java/org/eclipse/che/ide/ui/loaders/LoaderPresenter.java]: can be moved to localization

**Comment by vitaliy-guliy** [comment]
@TylerJewell Yes, sure. Trying to reduce the width.

**Comment by vitaliy-guliy** [comment]
dark theme
light theme

**Comment by vitaliy-guliy** [comment]
Everything is fixed.

**Comment by ashumilova** [comment]
+1

**Comment by vparfonov** [comment]
ok

**Pull request merged**